crates/model/src/order.rs

```diff
@@ -614,6 +614,9 @@ pub struct OrderMetadata {
     /// Does not take partial fill into account.
     #[serde(default, with = "u256_decimal")]
     pub solver_fee: U256,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<DecimalU256>")]
+    pub executed_solver_fee: Option<U256>,
```

Comment:
1. As Martin suggested this doesn't need be serialized because it's not supposed to show up in the api. If we want that, then it should be `#[serde(skip)]` .
2. If we want to have it serialized and use skip-if-none then you need to change it to `#[serde(default, skip_serializing_if = "Option::is_none")]`. There is an added `default` at the start. The current code fails to deserialize if the value is None. Unfortunately that doesn't seem to be caught by any test.

Reply:
I manually tested with `deserialization_and_back` test and it deserializes to `None` if missing (if I understood you well).
Anyway, I added `#[serde(skip)]`.